From: Joel Selvaraj <joelselvaraj.oss@gmail.com>
Date: Sat, 17 Sep 2022 00:47:27 -0500
Subject: tfa98xx: use devm_gpiod_put instead of deprecated devm_gpio_free

The devm_gpio_free is completely removed in 6.0. Check the below patch
https://lore.kernel.org/lkml/CAMRc=Mfqo19we+0ShbpCtbbsXgU6trKbwQR2Zg7LcAWSJ_834A@mail.gmail.com/T/
Temporarily convert gpio to gpio_desc and use devm_gpiod_put to release the gpio.
Eventually we will have to move to the new gpiod api.
---
 sound/soc/codecs/tfa98xx-downstream/tfa98xx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/codecs/tfa98xx-downstream/tfa98xx.c b/sound/soc/codecs/tfa98xx-downstream/tfa98xx.c
index 8a8b165..e54fcb2 100644
--- a/sound/soc/codecs/tfa98xx-downstream/tfa98xx.c
+++ b/sound/soc/codecs/tfa98xx-downstream/tfa98xx.c
@@ -3297,9 +3297,9 @@ static int tfa98xx_i2c_remove(struct i2c_client *i2c)
 	snd_soc_unregister_codec(&i2c->dev);
 #endif
 	if (gpio_is_valid(tfa98xx->irq_gpio))
-		devm_gpio_free(&i2c->dev, tfa98xx->irq_gpio);
+		devm_gpiod_put(&i2c->dev, gpio_to_desc(tfa98xx->irq_gpio));
 	if (gpio_is_valid(tfa98xx->reset_gpio))
-		devm_gpio_free(&i2c->dev, tfa98xx->reset_gpio);
+		devm_gpiod_put(&i2c->dev, gpio_to_desc(tfa98xx->reset_gpio));
 
 	mutex_lock(&tfa98xx_mutex);
 	list_del(&tfa98xx->list);
